### PR TITLE
Add Metric for Haproxy Backends Health

### PR DESCRIPTION
**Pull request by djquan**
This metric should allow services that depend on an Haproxy Backend to provide a health metric that reflects the status of the servers in that backend.  

Use case: Currently, we rely on a proxy that points to servers with varying weights. If a large percentage of these weights are not reachable, we'd like the health of that proxy to reflect that. 

@duelinmarkers @lollipopman

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -179,6 +179,11 @@ Here are all the types of checks currently implemented:
 - `big_brother_service` (`Metric::BigBrotherService`): Used in conjunction with [Big Brother](https://github.com/braintree/big_brother), reports health based on the overall health of another load balanced service.
   * service
 
+- `haproxy_backends_health` (`Metric::HaproxyBackendsHealth`): Checks UP/DOWN status across a set of servers in a Haproxy cluster and computes a weight based on how many are UP.
```
Comment (lollipopman) [inline README.md]: "and computes a weight based on how many are UP."

perhaps something like this insted:

"and computes a litmus weight based on the HAProxy weights of the servers which have an UP status"?

File: README.md [hunk 2]
```diff
@@ -179,6 +179,11 @@ Here are all the types of checks currently implemented:
...
+  * haproxy socket
+  * cluster
+  * timeout (defaults to 2s)
```
Comment (lollipopman) [inline README.md]: "* weight (0-100)"

File: README.md (hunk 1 above)
Comment (djquan) [reply README.md]: 👍 4b086b85d1e25a2d80c5e057d75e2440e8c5907d

File: README.md (hunk 2 above)
Comment (djquan) [reply README.md]: 👍  4b086b85d1e25a2d80c5e057d75e2440e8c5907d

Comment (djquan) [reply README.md]: (We realized we should ad weight to the arguments based on this comment, so that is done ad1032455d9cdb352fa0debf01c0e3e0064a65c2)

**Pull request merged**